src/main/java/edu/uw/edm/edmzuulproxy/certificateauthorizer/service/impl/CertificateAuthorizerServiceImpl.java

```diff
@@ -31,63 +33,64 @@
+        this.certificateAuthorizationRetriever = certificateAuthorizationRetriever;
         this.certificateAuthorizationRepository = certificateAuthorizationRepository;
     }
 
     @Override
     @Cacheable(cacheNames = "authorized-uri")
     public boolean isAllowedForUri(String certificateName, HttpMethod httpMethod, String uri, User user) {
-        log.trace("Checking if cert {} - user {} is allowed to access {} - {}", certificateName, user.getUsername(), httpMethod.name(), uri);
+        log.trace("Checking if cert {} - user {} is allowed to access {} - {}", certificateName, user, httpMethod.name(), uri);
 
-        final Iterable<CertificateAuthorization> byCertificateName = certificateAuthorizationRepository.findByCertificateName(certificateName);
+        final Iterable<CompiledCertificateAuthorization> byCertificateName = certificateAuthorizationRetriever.findByCertificateName(certificateName);
 
 
-        for (CertificateAuthorization certificateAuthorization : byCertificateName) {
-            if (isAuthorizedForRequest(certificateAuthorization, httpMethod, uri, user)) {
+        for (CompiledCertificateAuthorization certificateAuthorizationDAO : byCertificateName) {
+            if (isAuthorizedForRequest(certificateAuthorizationDAO, httpMethod, uri, user)) {
                 return true;
             }
         }
 
         return false;
     }
 
-    private boolean isAuthorizedForRequest(CertificateAuthorization certificateAuthorization, HttpMethod httpMethod, String uri, User user) {
+    private boolean isAuthorizedForRequest(CompiledCertificateAuthorization certificateAuthorizationDAO, HttpMethod httpMethod, String uri, User user) {
 
-        final boolean methodIsAuthorized = methodIsAuthorized(httpMethod, certificateAuthorization);
-        final boolean uriIsAuthorized = uriIsAuthorized(uri, certificateAuthorization);
-        final boolean uwGroupsMatch = uwGroupsMatch(user, certificateAuthorization);
+        final boolean authorized = methodIsAuthorized(httpMethod, certificateAuthorizationDAO) &&
+                uriIsAuthorized(uri, certificateAuthorizationDAO)
+                && uwGroupsMatch(user, certificateAuthorizationDAO);
 
-
-        final boolean authorized = methodIsAuthorized &&
-                uriIsAuthorized &&
-                uwGroupsMatch;
-
-        log.trace("method : {} , uri : {}, uwGroups: {} => result : {}", methodIsAuthorized, uriIsAuthorized, uwGroupsMatch, authorized);
         return authorized;
 
     }
 
-    private boolean uwGroupsMatch(User user, CertificateAuthorization certificateAuthorization) {
-        return certificateAuthorization.getUwGroups() == null || certificateAuthorization.getUwGroups().contains(UW_GROUPS_WILDCARD) || userIsMemberOfOneGroup(user, certificateAuthorization);
+    private boolean uwGroupsMatch(User user, CompiledCertificateAuthorization certificateAuthorization) {
+        return certificateAuthorization.getUwGroups() == null
```

Comment:
shouldn't be null

Reply:
I'm now checking for null ( or empty ) of all the mandatory fields in edu.uw.edm.edmzuulproxy.certificateauthorizer.service.impl.CertificateAuthorizationRetriever#toCompiledCertificateAuthorization )